Libplanet.Tests/Assets/FungibleAssetValueTest.cs

```diff
@@ -0,0 +1,306 @@
+            Assert.True(foo100a != bar200a);
+            Assert.NotEqual(bar100a, foo200a);
+            Assert.False(bar100a.Equals((object)foo200a));
+            Assert.False(bar100a == foo200a);
+            Assert.True(bar100a != foo200a);
+            Assert.NotEqual(foo100a, bar100a);
+            Assert.False(foo100a.Equals((object)bar100a));
+            Assert.False(foo100a == bar100a);
+            Assert.True(foo100a != bar100a);
+            Assert.NotEqual(foo100a, bar200a);
+            Assert.False(foo100a.Equals((object)bar200a));
+            Assert.False(foo100a == bar200a);
+            Assert.True(foo100a != bar200a);
+
+            Assert.False(foo100a.Equals(100));
+            Assert.False(foo200a.Equals(200));
+        }
+
+        [Fact]
+        public void Compare()
+        {
+            FungibleAssetValue foo100a = new FungibleAssetValue(FOO, 100);
+            FungibleAssetValue foo100b = new FungibleAssetValue(FOO, 100);
+            FungibleAssetValue foo200 = new FungibleAssetValue(FOO, 200);
+            FungibleAssetValue bar100 = new FungibleAssetValue(BAR, 100);
+
+            Assert.Equal(0, foo100a.CompareTo(foo100b));
+            Assert.Equal(0, foo100a.CompareTo((object)foo100b));
+            Assert.False(foo100a < foo100b);
+            Assert.True(foo100a <= foo100b);
+            Assert.False(foo100a > foo100b);
+            Assert.True(foo100a >= foo100b);
+
+            Assert.True(foo100a.CompareTo(foo200) < 0);
+            Assert.True(foo100a.CompareTo((object)foo200) < 0);
+            Assert.True(foo100a < foo200);
+            Assert.True(foo100a <= foo200);
+            Assert.False(foo100a > foo200);
+            Assert.False(foo100a >= foo200);
+
+            Assert.True(foo200.CompareTo(foo100b) > 0);
+            Assert.True(foo200.CompareTo((object)foo100b) > 0);
+            Assert.False(foo200 < foo100b);
+            Assert.False(foo200 <= foo100b);
+            Assert.True(foo200 > foo100b);
+            Assert.True(foo200 >= foo100b);
+
+            Assert.Throws<ArgumentException>(() => foo100a.CompareTo(bar100));
+            Assert.Throws<ArgumentException>(() => foo100a.CompareTo((object)bar100));
+            Assert.Throws<ArgumentException>(() => foo100a < bar100);
+            Assert.Throws<ArgumentException>(() => foo100a <= bar100);
+            Assert.Throws<ArgumentException>(() => foo100a > bar100);
+            Assert.Throws<ArgumentException>(() => foo100a >= bar100);
+
+            Assert.Throws<ArgumentException>(() => foo100a.CompareTo(100));
+        }
+
+        [Fact]
+        public void Negate()
+        {
+            FungibleAssetValue foo_3 = new FungibleAssetValue(FOO, -3);
+            FungibleAssetValue foo0 = new FungibleAssetValue(FOO);
+            FungibleAssetValue foo3 = new FungibleAssetValue(FOO, 3);
+
+            Assert.Equal(foo_3, -foo3);
+            Assert.Equal(foo3, -foo_3);
+            Assert.Equal(foo0, -foo0);
```

Comment:
Have we offer negative? I wonder when it can be negative.

Reply:
Sorry, I don't get it well what you mean… but if you mean “what does mean negative fungible asset values?” negative values purpose to represent debts.